torch/_inductor/scheduler.py

```diff
@@ -1654,6 +1767,91 @@ def __init__(self, nodes: List[ir.Operation]) -> None:
             }
         )
 
+    def get_runtime_dict(self, node):
+        total_node = 0
+        run_time_dict = {}
+        for n in node:
+            if isinstance(n, FusedSchedulerNode):
+                total_time, total_memory = 0, 0
+                buffer_op_name = [str(i) for i in n.snodes[0].node.origins]
+                buffer_op_name = "".join(buffer_op_name)
+                for subnode in n.snodes:
+                    stime, smemory = subnode.get_benchmark_runtime()
+                    total_time += stime
+                    total_memory += smemory
+                run_time_dict[n.get_name()] = [buffer_op_name, total_time, total_memory]
+            elif isinstance(n, GroupedSchedulerNode):
+                total_time, total_memory = 0, 0
+                buffer_op_name = [str(i) for i in n.snodes[0].node.origins]
+                buffer_op_name = "".join(buffer_op_name)
+                for subnode in n.snodes:
+                    stime, smemory = subnode.get_benchmark_runtime()
+                    total_time += stime
+                    total_memory += smemory
+                run_time_dict[n.get_name()] = [buffer_op_name, total_time, total_memory]
+            else:
+                if isinstance(n.node, ir._WaitKernel):
+                    run_time_dict[n.get_name()] = ["", 0, 0]
+                else:
+                    python_kernel = getattr(n.node, "python_kernel_name", None)
+                    if python_kernel is not None:
+                        buffer_op_name = n.node.python_kernel_name
+                    else:
+                        buffer_op_name = [str(i) for i in n.node.origins]
+                        buffer_op_name = "".join(buffer_op_name)
+
+                    benchmark_op = n.get_benchmark_runtime()
+                    run_time_dict[n.get_name()] = [buffer_op_name] + benchmark_op
+        return run_time_dict
+
+    def profile_nodes(self, node) -> Dict[str, List[Union[str, float, float]]]:
```

Comment:
ditto: let's put it in `utils.py`

Reply:
Let's have some comments on this, explaining that we need to broadcast from rank 0 (on each PP stage) to make sure every rank receives the same runtime estimation results.